lib/chef/provider/git.rb

```diff
@@ -154,6 +154,15 @@ def find_current_revision
         sha_hash?(result) ? result : nil
       end
 
+      def already_on_target_branch?
+        current_branch = git("rev-parse", "--abbrev-ref", "HEAD", cwd: cwd, returns: [0, 128]).stdout.strip
+        if new_resource.checkout_branch
+          current_branch == new_resource.checkout_branch
+        else
+          current_branch == new_resource.revision
+        end
```

Comment:
```suggestion
        current_branch == (new_resource.checkout_branch || new_resource.revision)
```

This could be made more succinct.

Reply:
ohh nice. Pushed it